.github/workflows/ci.yml

```diff
@@ -150,47 +93,85 @@ jobs:
         OBJC_DISABLE_INITIALIZE_FORK_SAFETY: "YES"
       with:
         command: test
-        args: -j 2 --workspace --all-features --examples --benches --bins
-
-    - name: clippy
+        args: --workspace --all-features --examples --benches --bins
+    
+    - name: tests w/o default features
       uses: actions-rs/cargo@v1
-      if: matrix.os == 'ubuntu-latest' && matrix.rust=='stable'
+      timeout-minutes: 30
+      env:
+        # iroh-util/src/lock.rs#test_locks uses a fork() command & fails only in
+        # GHA if this is not present
+        # https://twitter.com/steipete/status/921066430220652544?lang=en
+        # https://blog.phusion.nl/2017/10/13/why-ruby-app-servers-break-on-macos-high-sierra-and-what-can-be-done-about-it/
+        OBJC_DISABLE_INITIALIZE_FORK_SAFETY: "YES"
       with:
-        command: clippy
-        args: --workspace --all-features --tests --benches --examples --bins --all-targets -- -D warnings
+        command: test
+        args: --workspace --no-default-features --examples --benches --bins
 
-    - name: build release
-      uses: actions-rs/cargo@v1
-      if: matrix.rust=='stable' && github.ref_name=='main'
-      with:
-        command: build
-        args: --profile ci
+  build_release:
+    name: Build release binaries
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    needs: build_and_test
+    if: ${{ always() && github.ref_name=='main' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+        rust: [stable]
+        experimental: [false]
+        include:
+          - os: ubuntu-arm-latest
+            release-os: linux
+            release-arch: aarch64
+            runner: [self-hosted, linux, ARM64]
+          - os: ubuntu-latest
+            release-os: linux
+            release-arch: amd64
+            runner: [self-hosted, linux, X64]
+          - os: macos-latest
+            release-os: darwin
+            release-arch: x86_64
+            runner: [self-hosted, macOS, X64]
+          # - os: windows-latest
+          #   release-os: windows
+          #   release-arch: amd64
+          #   runner: [windows-latest]
+    env:
+      RUST_BACKTRACE: full
+      # Force not building debuginfo to save space on disk.
+      RUSTFLAGS: "-C debuginfo=0"
```

Comment:
For releases we probably want to build debuginfo and use split debuginfo and make it available in the release as well.  But maybe that's also something that is better done later.